code/_onclick/hud/human.dm

```diff
@@ -359,38 +359,39 @@
 
 	var/mob/screenmob = viewer || H
 
-	if(screenmob.hud_used.hud_shown)
-		if(H.s_store)
-			H.s_store.screen_loc = ui_sstore1
-			screenmob.client.screen += H.s_store
-		if(H.wear_id)
-			H.wear_id.screen_loc = ui_id
-			screenmob.client.screen += H.wear_id
-		if(H.belt)
-			H.belt.screen_loc = ui_belt
-			screenmob.client.screen += H.belt
-		if(H.back)
-			H.back.screen_loc = ui_back
-			screenmob.client.screen += H.back
-		if(H.l_store)
-			H.l_store.screen_loc = ui_storage1
-			screenmob.client.screen += H.l_store
-		if(H.r_store)
-			H.r_store.screen_loc = ui_storage2
-			screenmob.client.screen += H.r_store
-	else
-		if(H.s_store)
-			screenmob.client.screen -= H.s_store
-		if(H.wear_id)
-			screenmob.client.screen -= H.wear_id
-		if(H.belt)
-			screenmob.client.screen -= H.belt
-		if(H.back)
-			screenmob.client.screen -= H.back
-		if(H.l_store)
-			screenmob.client.screen -= H.l_store
-		if(H.r_store)
-			screenmob.client.screen -= H.r_store
+	if(screenmob.hud_used)

```

Comment:
why fix it this way instead of doing the check just before calling show_hud() in build_hand_slots()? It makes more sense to have it in build_hand_slots() since we only need to call show_hud() in there when we already have one set. Otherwise creating a hud will call show_hud() twice.